CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.12)
 
-project(espressopp LANGUAGES CXX)
+project(espressopp LANGUAGES C CXX)
```

Comment:
Instead of adding `C` here, I would add `enable_language(C)` (with a small comment) where `find_package(HFD5)` is called.

Reply:
What is the benefit of this?

Reply:
I think that would make it more clear that the only reason we  need `C` is due to `HDF5`.

Reply:
..., `C` is only needed due to a compile test within `FindHDF5.cmake`. The file extension is `c` and, therefore, a c compiler is needed. If we change that to `cpp` we do not need `C` as a language in our project. However, we have to ship our own slightly modified version of `FindHDF5.cmake`. At the same time this would allow us to use exported targets, since we can build upon the CMake 3.20 version. @junghans What do you think?

Reply:
I like that! And we should report that issue to cmake upstream as well!

Reply:
There was already an attempt to do this:
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4534
TL;DR Enable C language if you use C language bindings.